walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/servlets/FindMeetingQuery.java

```diff
@@ -0,0 +1,151 @@
+                return Arrays.asList(TimeRange.WHOLE_DAY);
+            } 
+            //Checks if optional attendees can attend open times, but will not affect times made for mandatory attendees
+            if (checkAttendees(eventAttendees, request.getOptionalAttendees()) && !request.getAttendees().isEmpty())
+            {
+                return checkIfOptionalAttendeesCanAttend(availableTime,whenEvent);
+            }
+        }
+        return availableTime;
+    }
+
+    private void addFirstOpenTimes(long meetingDuration, Collection<TimeRange> availableTime, TimeRange whenEvent)
+    {
+        TimeRange frontTime1= TimeRange.fromStartEnd(TimeRange.START_OF_DAY,whenEvent.start(),false);
+        TimeRange backTime1 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+        addFrontAndBackTimesToList(meetingDuration, frontTime1, backTime1, availableTime);
+    }
+    
+    /** Checks if available times overlap event times to make available times around the event */
+    private void checkOverlapsInPotentialTimesWithEventTimes(Collection<TimeRange> availableTime,TimeRange whenEvent,
+        long meetingDuration)
+    {
+        Collection<TimeRange> removeList = new ArrayList<TimeRange>();
+        Collection<TimeRange> addList = new ArrayList<TimeRange>();
+        for (TimeRange openTime : availableTime)
+        {
+            if(openTime.overlaps(whenEvent))
+            {
+                if(openTime.contains(whenEvent))
+                {
+                    removeList.add(openTime);
+                    TimeRange frontTime2 = TimeRange.fromStartEnd(openTime.start(),whenEvent.start(),false);
+                    TimeRange backTime2 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+                    addFrontAndBackTimesToList(meetingDuration, frontTime2, backTime2,addList);
+                }
+                else if (whenEvent.start() < openTime.start())
+                {
+                    removeList.add(openTime);
+                    TimeRange overlapTime = TimeRange.fromStartEnd(whenEvent.end(),openTime.end(),false);
+                    if (overlapTime.duration() >= meetingDuration)
+                    {
+                        addList.add(overlapTime);
+                    }
+                }
+            }
+        }
+        availableTime.removeAll(removeList);
+        availableTime.addAll(addList);    
+    }
+
+    private Collection<TimeRange> checkIfOptionalAttendeesCanAttend(Collection<TimeRange> availableTime,TimeRange whenEvent)
+    {
+        Collection<TimeRange> timeWithOptional = new ArrayList<TimeRange>(); 
+        for(TimeRange currentOpenTime: availableTime)
+        {
+            if(!currentOpenTime.overlaps(whenEvent))
+            {
+                timeWithOptional.add(currentOpenTime);
+            }
+        }
+        if (timeWithOptional.isEmpty())
+        {
+            return availableTime;
+        }
+        return timeWithOptional;
```

Comment:
You could replace this branching with a ternary operator to make it more succinct:

return timeWithOptional.isEmpty() ? availableTime : timeWithOptional;